src/ScottPlot5/ScottPlot5/Plottables/Scatter.cs

```diff
@@ -3,7 +3,7 @@
 
 namespace ScottPlot.Plottables;
 
-public class Scatter(IScatterSource data) : IPlottable, IHasLine, IHasMarker, IHasLegendText
+public class Scatter(IScatterSource data) : IPlottable, IHasLine, IHasMarker, IHasLegendText, IDataSource, IGetNearest
```

Comment:
Why does IPlottable Scatter implement an IDataSource when it has its own?

Reply:
I think I realized, Scale offset is the responsibility of Scatter, and DataSource itself can't do it (Previous design you saved).
I guess someday later this can all be fixed. I think any plottable would love to have a DataSouce with Scale and Offset support. And this will be easy to do with the helper methods of this PR.

Reply:
Full disclosure, I've only been working with signal and scatter, I have not yet looked at any of the other plants as I have not needed them. But I noticed that the ISignalSource differs from some other sources in that, it specifies the offsets. 

Now, I have no idea how other people are using this, but to me the offset and the scaling are more UI (plottable) than data (ISignalSource, IScatterSource, etc). 

The GetNearest methods All have to do with mouse coordinates, as such they are UI related. This is why IDataSource has GetX and GetXScaled, to provide a mechanism to get the UI coordinate and a separate mechanism to get the data coordinate. 

I think the flub here was that the signals have scaling at all and that GetNearest (a ui function) was implemented at the data level. I understand that this is very much a zoomed out and hindsight statement, I am making here (and am not criticizing). But I think that it may take a major overhaul to fix those nuances, which may break consumers.

To me, the sources should be data-focused. No offsets, no scaling. GetNearest would be fine I think, but callers would have to use an unscaled coordinate, as it loses offset and scaling information required to reach a correct result. 

IGetNearest  (or IPlottable) would then have the GetNearest functions, as well as an IDataSource property and the Get[XY]Scaled methods.